components/cards/ClaimAction.tsx

```diff
@@ -13,78 +22,166 @@ export type ClaimActionProps = SidebarProps & {
-    //   filter,
-    //   receipt.blockNumber,
-    //   receipt.blockNumber
-    // );
-    // let licenseId = res[0].args[0];
-    // return licenseId;
-    return "";
+    const baseCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase1Coord.x,
+      claimBase1Coord.y
+    );
+    const destCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase2Coord.x,
+      claimBase2Coord.y
+    );
+    let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
+      // FIXME: it should be 'BN' type, waiting for js-geo-web-coordinate upgrade.
+      (v: any) => {
+        return BigNumber.from(v.toString(10));
+      }
+    );
+    if (path.length == 0) {
+      path = [BigNumber.from(0)];
+    }
+
+    if (!displayNewForSalePrice) {
+      throw new Error(
+        `displayNewForSalePrice is invalid: ${displayNewForSalePrice}`
+      );
+    }
+
+    console.log(actionData);
+
+    const claimData = ethers.utils.defaultAbiCoder.encode(
+      ["uint64", "uint256[]"],
+      [BigNumber.from(baseCoord.toString(10)), path]
+    );
+
+    const actionDataToPassInUserData = ethers.utils.defaultAbiCoder.encode(
+      ["uint256", "bytes"],
+      [ethers.utils.parseEther(displayNewForSalePrice), claimData]
+    );
+
+    let userData;
+
+    let txn;
+
+    const sf = await sfInstance(NETWORK_ID, provider);
+
+    const ethx = await sf.loadSuperToken(paymentTokenAddress);
+
+    const existingFlow = await sf.cfaV1.getFlow({
+      superToken: paymentTokenAddress,
+      sender: account,
+      receiver: auctionSuperApp.address,
+      providerOrSigner: provider as any,
+    });
+
+    const approveOperation = await ethx.approve({
+      receiver: auctionSuperApp.address,
+      amount: ethers.utils.parseEther(displayNewForSalePrice).toString(),
+    });
+
+    const networkFeeRatePerSecond = fromValueToRate(
+      ethers.utils.parseEther(displayNewForSalePrice),
+      perSecondFeeNumerator,
+      perSecondFeeDenominator
+    );
+
+    const newFlowRate = networkFeeRatePerSecond;
+
+    const signer = provider.getSigner() as any;
+
+    if (existingFlow.flowRate !== "0") {
+      // update an exisiting flow
+      console.log("Updting an exisiting flow: ");
+
+      userData = ethers.utils.defaultAbiCoder.encode(
+        ["uint8", "bytes"],
+        [Action.CLAIM, actionDataToPassInUserData]
+      );
+
+      const updateFlowOperation = await sf.cfaV1.updateFlow({
+        flowRate: BigNumber.from(existingFlow.flowRate)
+          .add(newFlowRate)
+          .toString(),
+        receiver: auctionSuperApp.address,
+        superToken: paymentTokenAddress,
+        userData,
+      });
+
+      txn = await sf
+        .batchCall([approveOperation, updateFlowOperation])
+        .exec(signer);
+
+      console.log(txn);
+    } else {
+      // create a new flow
+      console.log("Creating a new flow: ");
+
+      userData = ethers.utils.defaultAbiCoder.encode(
+        ["uint8", "bytes"],
+        [Action.CLAIM, claimData]
```

Comment:
Ok, all this abi.encode stuff is a pain to troubleshoot, but I was able to get a claim to succeed after fixing this: `claimData` -> `actionDataToPassInUserData`.

The claim succeeded, but the event query failed below https://github.com/Geo-Web-Project/cadastre/pull/126/files#diff-77130f94448dec75dfc66a3dc0c3af3f07a3329cf0a300be36bcb042f7f8fd64R176

Reply:
@codynhat 

500 error?

<img width="673" alt="subgraph commits 500 error" src="https://user-images.githubusercontent.com/31362988/169448604-91adb05f-b384-47da-8e52-b2a3eeaa45e6.png">

Reply:
Hmm, I didn't get that. Just an error about `res[0]` being undefined.

Reply:
I created a parcel successfully with 500 error, and now I can update the parcel but still get this 500 error:

<img width="1239" alt="Screen Shot 2022-05-20 at 12 50 07" src="https://user-images.githubusercontent.com/31362988/169452838-22d8a546-1a7d-4949-b851-18b709f1804f.png">

**BTW**: could you retrieve it from the geo web subgraph?

Reply:
I claimed a new parcel, and I can not get the `res[0] is undefined` error (got an array with 4 elements instead):

<img width="1464" alt="Screen Shot 2022-05-20 at 15 54 12" src="https://user-images.githubusercontent.com/31362988/169481391-ad07ac32-3856-4c34-a29f-45bb3b03384e.png">

Reply:
Yep, I see some of your claims. I am not able to reproduce the `res[0] is undefined`. Let's monitor to see if it appears again. I am also getting the 500. But let's mark this as complete and address that issue in https://github.com/Geo-Web-Project/cadastre/issues/108

Could you just clean up some of the console.logs? Then I will approve and merge.

Reply:
Done